docs/developer-guide/ksqldb-clients/java-client.md

```diff
@@ -0,0 +1,473 @@
+---
+layout: page
+title: Java Client for ksqlDB
+tagline: Java client for ksqlDB
+description: Send requests to ksqlDB from your Java app
+keywords: ksqlDB, java, client
+---
+
+ksqlDB ships with a lightweight Java client that allows you to easily send requests to a ksqlDB server
+from within your Java application, as an alternative to using the [REST API](../api.md).
+The client currently supports pull and push queries as well as inserting new rows of data into existing ksqlDB streams.
+Soon the client will also support persistent queries and admin operations such as listing streams, tables, and topics. 
```

Comment:
Generally, we don't want to make promises about future features in docs, but It's probably fine in this case.